src/Servers/Kestrel/Core/src/Internal/KestrelServerImpl.cs

```diff
@@ -302,14 +297,16 @@ private async Task RebindAsync()
                     return;
                 }
 
+                Debug.Assert(Options.ConfigurationLoader != null); // TODO - Can ConfigurationLoader be null here?
```

Comment:
RebindAsync will never be called unless there's a ConfigurationLoader to get a reload token from to begin with. The setter for KestrelServerOptions.ConfigurationLoader is public, so I guess it's *technically* possible that it becomes null. KestrelConfigurationLoader doesn't even have a public ctor though, so I don't find it very likely.